R/utils.R

```diff
@@ -230,3 +230,68 @@ padding_lst <- function(ft, indents) {
     flextable::padding(ft, s, 1, padding.left = (indents[s] + 1) * 10)
   }, seq_len(length(indents)), ft)
 }
+
+#' Split a text block into smaller blocks with a specified number of lines.
+#'
+#' This function takes a block of text and divides it into smaller blocks, each containing
+#' a specified number of lines.
+#'
+#' @param block_text A character vector containing the input block of text.
+#' @param n The number of lines per block.
+#'
+#' @return A list of character vectors, where each element is a smaller block of text
+#'         containing 'n' lines. If the input block of text has fewer lines than 'n', the
+#'         entire block is returned as a single element list.
+#'
+#' @keywords internal
+split_text_into_blocks <- function(block_text, n) {
+  lines <- strsplit(block_text, "\n")[[1]]
+  num_lines <- length(lines)
+
+  if (num_lines <= n) {
+    return(list(block_text))
+  }
+
+  num_blocks <- ceiling(num_lines / n)
+  blocks <- vector("list", length = num_blocks)
+
+  for (i in 1:num_blocks) {
+    start <- (i - 1) * n + 1
+    end <- min(i * n, num_lines)
+    block <- paste(lines[start:end], collapse = "\n")
+    blocks[[i]] <- block
+  }
+
+  return(blocks)
+}
+
+#' Extract a Field from YAML and Optionally Retrieve Names from a List
+#'
+#' This function parses a YAML text and extracts the specified field. It provides
+#' the option to retrieve the names of elements from a list if the field contains a list.
+#'
+#' @param yaml_text A character vector containing the YAML text.
+#' @param field_name The name of the field to extract.
+#' @param check_list Logical, indicating whether to check if the result is a list
+#'                   and retrieve the names of list elements. Default is TRUE.
+#'
+#' @return If `check_list` is TRUE and the result is a list, it returns the names of
+#'         elements in the list; otherwise, it returns the extracted field.
+#'
+#' @keywords internal
+reverse_yaml_field <- function(yaml_text, field_name, check_list = TRUE) {
```

Comment:
There is a file in this package called `yaml_utils.R`.

Reply:
thanks.